### PR TITLE
community: add DynamoDB store

### PR DESCRIPTION
**Pull request by russell-dot-js**
- **Description:** added a DynamoDb implementation of ByteStore. I'm putting this PR up for feedback before hardening with unit tests because the requirement to implement yield_keys with a prefix string meant either always performing a full table scan, or having a requirement to use a compound key. The DynamoDb MessageHistory implementation does not require having a compound key, so this could be a point of contention for those who follow a single table design. 
  - **Issue:** N/A
  - **Dependencies:** N/A
  - **Twitter handle:** kingsmendv

### CODE REVIEW

File: libs/community/langchain_community/storage/dynamodb.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            response = self.client.batch_get_item(
+                RequestItems={
+                    self.table.name: {
+                        "Keys": [self.make_key_dict(key) for key in chunk],
+                        "AttributesToGet": ["Bytes"]
+                    }
+                }
+            )
+
+            for item in response["Responses"][self.table.name]:
```
Comment (russell-dot-js) [inline libs/community/langchain_community/storage/dynamodb.py:144]: todo: handle unprocessed keys, return key to ensure order

**Comment by baskaryan** [comment]
hey @russell-dot-js, is this pr ready for review?

**Comment by russell-dot-js** [comment]
> hey @russell-dot-js, is this pr ready for review?

Check out my PR description - I'm looking for direction from a contributor who's able to make the call about whether or not the design is acceptable (forces compound key) before hardening with tests

**Comment by baskaryan** [comment]
> > hey @russell-dot-js, is this pr ready for review?
> 
> Check out my PR description - I'm looking for direction from a contributor who's able to make the call about whether or not the design is acceptable (forces compound key) before hardening with tests

ah got. could we let users specify during init if they're using compound key, and if that's not specified then prefix arg isn't supported?

**Comment by eyurtsev** [comment]
@russell-dot-js sorry for taking so long to review. Happy to merge

You can use the standard test suite to verify that the implementation works:

https://github.com/langchain-ai/langchain/pull/23876/files#diff-9b138a87cb000c2ee0d8d47389d54a5edd33403a1c8165099e619dccc84e9efaR77

**Comment by efriis** [comment]
closing as stale - let me know if that's incorrect!

**Pull request closed without merging**